src/accelerate/accelerator.py

```diff
@@ -205,13 +213,23 @@ def __init__(
         megatron_lm_plugin: MegatronLMPlugin = None,
         rng_types: Optional[List[Union[str, RNGType]]] = None,
         log_with: Optional[List[Union[str, LoggerType, GeneralTracker]]] = None,
+        project_dir: Optional[Union[str, os.PathLike]] = None,
+        save_config: Optional[SaveConfiguration] = None,
         logging_dir: Optional[Union[str, os.PathLike]] = None,
         dispatch_batches: Optional[bool] = None,
         even_batches: bool = True,
         step_scheduler_with_optimizer: bool = True,
         kwargs_handlers: Optional[List[KwargsHandler]] = None,
         dynamo_backend: Union[DynamoBackend, str] = None,
     ):
+        if save_config is not None:
+            self.save_configuration = save_config
+        else:
+            self.save_configuration = SaveConfiguration()
+        if project_dir is not None and self.project_dir is None:
+            self.save_configuration.project_dir = project_dir
+        if self.project_dir is not None and logging_dir is None:
```

Comment:
Let's also deprecate logging_dir to include it in the SaveConfig? Wdyt?

Reply:
I was debating on that, if we did so I think it'd be better to name it `ProjectConfiguration` instead since it's all relative to the total "project" at that point. But can make adjustments for it. The only "con" I see is given the conversation above for the user the code would change, but the end-result would not so I think it'd be fine